nav2_common/cmake/nav2_package.cmake

```diff
@@ -37,7 +37,7 @@ macro(nav2_package)
   endif()
 
   if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    add_compile_options(-Wall -Wextra -Wpedantic -Werror -Wdeprecated -fPIC -Wshadow -Wnull-dereference)
```

Comment:
Readd

Reply:
Removing -Wnull-dereference temporarily because of https://stackoverflow.com/questions/77436264/potential-null-deference-warning

Reply:
Removing Wshadow temporarily because of https://github.com/BehaviorTree/BehaviorTree.CPP/issues/811